### DISCUSSION TITLE
Headless pi shuts down after program change

### DISCUSSION THREAD

**Discussion by StewRat83** [GeekFunkLabs/fluidpatcher#81] [Q&A]
Firstly, thanks so much for making this in the first place - I'm looking forward to using it, and installation has been a breeze compared with other attempts to do something like this. So great work!
The problem I am having is likely to be my fault as I have no real experience with midi control, but here it is ...
Started on 32 bit Lite on a Pi Zero, but problem is same on a new 64 bit Lite Pi install.
Keyboard/controller is a Nektar Impact LX49+
On startup via 'python3 headlesspi.py' or './headlesspi.py' I could play instrument 1 piano all day I think.
Button 1 on the keyboard appears to send CC21 by default so I could immediately move down through the instruments, but same problem after assigning CC21 and CC22 to buttons 7&8 so I can navigate up and down through the patches.
Soon after changing patch - sometimes immediately, sometimes I can play a few notes first, whether I go up or down, 1 patch or several - The Pi shuts down and powers off. The variability doesn't feel like it is the built in shutdown in action but it maybe.
 
Example 1 - I clicked down to patch 23 and waited with no key presses - off.
============================
stewart@pisynth:~ $ python3 headlesspi.py
Connection is already subscribed
Connection is already subscribed
Loading bank 'bank1.yaml' ..
Bank loaded.
Selected patch 1/26: Piano
Selected patch 26/26: Power Kit
Selected patch 25/26: Standard Kit
Selected patch 24/26: SFX
Selected patch 23/26: NewAge
Shutting down..
Connection to pisynth.local closed by remote host.
Connection to pisynth.local closed.
============================

Example 2 - I clicked up 1 patch and immediately it was off.
============================
stewart@pisynth:~ $ python3 headlesspi.py
Connection is already subscribed
Connection is already subscribed
Connection is already subscribed
Loading bank 'bank1.yaml' ..
Bank loaded.
Selected patch 1/26: Piano
Selected patch 2/26: Rhodes
Shutting down..
Connection to pisynth.local closed by remote host.
Connection to pisynth.local closed.
============================

There is a slightly similar Zero shut down thread but not related to patch changes, and the --interactive option doesn't seem to do anything these days.

Any suggestions welcome.

Thanks

Stewart

**Comment by StewRat83** [comment]
So, as is often the case, after an evening futzing around with this to no avail, I post a question and then (I think) answer it for myself minutes later.
Looking at the keyboard documentation (which is only slowly starting to make any sense!) I note that the default assignment type for the buttons I have been using is 'MIDI CC Toggle' which I guess is like holding a CC value down to trigger the shutdown.
Changing this to 'MIDI CC Trigger/Release' seems to have solved the problem.
Feel free to delete all this if it is just flotsam, or leave it in case it helps anyone else in future.

**Comment by albedozero** [comment]
I was just typing a reply :) -- This is documented in the troubleshooting section [here](https://github.com/albedozero/fluidpatcher/blob/master/troubleshooting.md#when-i-switch-patches-using-the-buttons-on-my-controller-the-pi-shuts-down).
